File: XH/exam_postprocess/Interface.py

from .exam_classify import ExamClassify
from .exam_analyse import ExamAnalyse

class InterfaceAnalyse():
    def __init__(self, model_path, gpu_id):
        super(InterfaceAnalyse, self).__init__()
        self.model = ExamClassify(model_path, gpu_id)
        self.exam_analyse = ExamAnalyse()



    def analyse_imgs(self, imgs):
        classifyes = self.model.Recognition(imgs)
        #print(classifyes)
        return self.exam_analyse.analyse(classifyes)
    def set_spend_time(self, spend_t):
        self.exam_analyse.set_spend_time(spend_t)
    def set_continue_time(self, continue_t):
        self.exam_analyse.set_continue_time(continue_t)

File: exam_postprocess/detect/Interface.cpp

#include "Interface.h"

InterfaceAnalyse::InterfaceAnalyse(std::string models_path){
	
	examClassify = std::make_shared<ExamClassify>(models_path);
	
}

void InterfaceAnalyse::set_spend_time(int spend_t) {
	exam_analyse.set_spend_time(spend_t);
}

void InterfaceAnalyse::set_continue_time(int continue_t) {
	exam_analyse.set_continue_time(continue_t);
}

bool InterfaceAnalyse::analyse_imgs(std::vector<cv::Mat> &imgs) {
	
	std::vector<int> classifyes;
	std::cout<<"classify:"<<std::endl;
	for(int i = 0; i < imgs.size(); i++){
		Classify classify;
		int status = examClassify->Recognition(imgs[i], classify);
		classifyes.push_back(classify.class_id);
		std::cout<<classify.class_id<<" ";
	}
	std::cout<<std::endl;
	return exam_analyse.analyse(classifyes);

}